rest/server_test.go

```diff
@@ -29,131 +29,155 @@ func (m *mockSecretService) WriteSecret(key string, value string, password strin
-	req.Header.Set("X-Cipher", "password")
-	client := http.Client{
-		Timeout: 5 * time.Second,
+func TestSecretRestAPI_Get(t *testing.T) {
+	type mockInput struct {
+		key      string
+		password string
 	}
-	res, err := client.Do(req)
-	responseData, _ := ioutil.ReadAll(res.Body)
-	assert.Equal(t, "value", string(responseData))
-	assert.Equal(t, http.StatusOK, res.StatusCode)
-	require.NoError(t, err)
-
-}
-func TestSecretRestAPI_Get_Error(t *testing.T) {
-	serverPort := strconv.Itoa(rand.Intn(20000-10000) + 10000)
-	mockSecretService := newMockSecretService()
-	mockSecretService.On("ReadSecret", "key", "password").Return("", errors.New("error"))
-	srv := NewSecretRestAPI(mockSecretService, serverPort)
-	serverCtx, serverCancel := context.WithCancel(context.Background())
-	defer serverCancel()
-	go func() {
-		srv.Start(serverCtx)
-	}()
-	time.Sleep(1 * time.Second)
-	req, _ := http.NewRequest(http.MethodGet, fmt.Sprintf("http://localhost:%s?getter=key", serverPort), nil)
-	req.Header.Set("X-Cipher", "password")
-	client := http.Client{
-		Timeout: 5 * time.Second,
+	type mockOutput struct {
+		result string
+		err    error
 	}
-	res, err := client.Do(req)
-	responseData, _ := ioutil.ReadAll(res.Body)
-	assert.Equal(t, "error\n", string(responseData))
-	assert.Equal(t, http.StatusNotFound, res.StatusCode)
-	require.NoError(t, err)
-
-}
-func TestSecretRestAPI_Post_Success(t *testing.T) {
-	serverPort := strconv.Itoa(rand.Intn(20000-10000) + 10000)
-	mockSecretService := newMockSecretService()
-	mockSecretService.On("WriteSecret", "key", "value", "password").Return(nil)
-	srv := NewSecretRestAPI(mockSecretService, serverPort)
-	serverCtx, serverCancel := context.WithCancel(context.Background())
-	defer serverCancel()
-	go func() {
-		srv.Start(serverCtx)
-	}()
-	time.Sleep(1 * time.Second)
-	jsonBody := []byte(`{"getter" : "key", "value" : "value"}`)
-	bodyReader := bytes.NewReader(jsonBody)
-
-	req, err := http.NewRequest(http.MethodPost, fmt.Sprintf("http://localhost:%s?getter=key", serverPort), bodyReader)
-	req.Header.Set("X-Cipher", "password")
-
-	client := http.Client{
-		Timeout: 5 * time.Second,
+	testCases := []struct {
+		name          string
+		mockInput     mockInput
+		mockOutput    mockOutput
+		expectedCode  int
+		expectedBody  string
+		expectedError error
+	}{
+		{
+			name: "successfully runs server",
+			mockInput: mockInput{
+				key:      "key",
+				password: "password",
+			},
+			mockOutput: mockOutput{
+				result: "value",
+				err:    nil,
+			},
+			expectedCode:  http.StatusOK,
+			expectedBody:  "value",
+			expectedError: nil,
+		},
+		{
+			name: "Returns error when fails to process mocked method",
+			mockInput: mockInput{
+				key:      "key",
+				password: "password",
+			},
+			mockOutput: mockOutput{
+				result: "",
+				err:    errors.New("error"),
+			},
+			expectedCode:  http.StatusNotFound,
+			expectedBody:  "error\n",
```

Comment:
No matter success or error a response body should be in the same format, in our case it is json.